### PR TITLE
Fix the bodyStartPos in the ParseSIPMessage Method. Fix the SIPUserAgent.OnCallHungup Event was not triggered when hanging up

### PR DESCRIPTION
**Pull request by danch15**
Fix the bodyStartPos in the ParseSIPMessage Method when the FromName or ToName is Chinese

### CODE REVIEW

**Comment by danch15** [comment]
#779 Call a number then cancel immediately, and then make an incoming call by other device, then answered. The SIPUserAgent.OnCallHungup Event was not triggered when hanging up.

**Review by sipsorcery — CHANGES_REQUESTED** [review]
Be great to see a unit test for this case.

File: src/core/SIP/SIPMessageBuffer.cs [hunk 1]
```diff
@@ -86,83 +86,85 @@ public string RawMessage
...
+                        sipMessage.SIPHeaders = SIPHeader.SplitHeaders(headerString);
+                    }
+                    else
+                    {
+                        string headerString = message.Substring(endFistLinePosn + 2, endHeaderPosn - endFistLinePosn - 2);
+                        sipMessage.SIPHeaders = SIPHeader.SplitHeaders(headerString);
+
+                        if (message.Length > endHeaderPosn + 4)
+                        {
+                            var bodyStartPos = sipEncoding.GetByteCount(headerString) + sipEncoding.GetByteCount(sipMessage.FirstLine) + 2 + 4;
```
[the remark this answers is not in the document]
Comment (danch15) [reply src/core/SIP/SIPMessageBuffer.cs:155]: string sipFromName = “呼叫方名字”;

To: "被叫方名字" <sip:5030004203@172.17.3.4:5060>;tag=LKFBNRWYOU
From: "呼叫方名字" <sip:57660000@172.17.2.2>;tag=DTLKNZCUCG

I am sorry I forget the new '2', but it must exist, your just test it.

**Comment by danch15** [comment]
I don't wanna pull request of the "src/core/SIP/SIPMessageBuffer.cs" and "src/core/SIP/Channels/SIPUDPChannel.cs" files, I just comit it to my fork, I'm not familiar with GitHub, Sorry. How can I cancel the pull request of these two files?

**Comment by danch15** [comment]
What the ???

**Comment by sipsorcery** [comment]
> What the ???

Closed the PR because you didn't want it merged into this repo.

**Comment by danch15** [comment]
> > What the ???
> 
> Closed the PR because you didn't want it merged into this repo.

https://github.com/sipsorcery-org/sipsorcery/issues/848

**Pull request closed without merging**